hbase-server/src/main/java/org/apache/hadoop/hbase/MetaRegionLocationCache.java

```diff
@@ -91,8 +92,8 @@ public MetaRegionLocationCache(ZKWatcher zkWatcher) {
     ThreadFactory threadFactory = new ThreadFactoryBuilder().setDaemon(true).build();
     RetryCounterFactory retryFactory = new RetryCounterFactory(Integer.MAX_VALUE,
       SLEEP_INTERVAL_MS_BETWEEN_RETRIES, SLEEP_INTERVAL_MS_MAX);
-    threadFactory.newThread(() -> loadMetaLocationsFromZk(retryFactory.create(), ZNodeOpType.INIT))
-      .start();
+    threadFactory.newThread(Context.current()
```

Comment:
I suppose this is used to pass the span in the current scope to the thread, so it will be recorded as parent span of later spans created in the thread. But what if the current span is ended before the spans created in the thread end?

Reply:
This is a very good point. Let me consider this more carefully.